### PR TITLE
CMP: introduce v3 and implement hashAlg field of certConf

### PR DESCRIPTION
**Pull request by DDvO**
* CMP: introduce version 3, while version 2 stays the default
* CMP: implement optional `hashAlg` field of `certConf` CMPv3 message

##### Checklist

- [x] documentation is added or updated

### CODE REVIEW

**Review by t8m — APPROVED** [review]

File: include/openssl/cmp.h.in [hunk 1]
```diff
@@ -36,7 +36,9 @@ use OpenSSL::stackhash qw(generate_stack_macros);
 extern "C" {
 #  endif
 
-#  define OSSL_CMP_PVNO 2
+#  define OSSL_CMP_PVNO 2 /* still the default */
```
Comment (slontis) [inline include/openssl/cmp.h.in:39]: _DEFAULT might of been better in hindsight..

Comment (t8m) [reply include/openssl/cmp.h.in:39]: Unfortunately that define is already present in current API.

File: include/openssl/cmp.h.in [hunk 2]
```diff
@@ -36,7 +36,9 @@ use OpenSSL::stackhash qw(generate_stack_macros);
...
+#  define OSSL_CMP_PVNO_2 OSSL_CMP_PVNO
```
Comment (t8m) [inline include/openssl/cmp.h.in:40]: What about:
```suggestion
#  define OSSL_CMP_PVNO_ 2 2
#  define OSSL_CMP_PVNO OSSL_CMP_PVNO_2 /* v2 is the default */
```

**Review by t8m — APPROVED** [review]

**Review by paulidale — APPROVED** [review]

**Comment by hlandau** [comment]
Merged. Thank you.

**Comment by DDvO** [comment]
> Merged. Thank you.

This merge was done in perfect time
since yesterday the CMP-Updates IETF draft introducing CMPv3 got its IESG approval :+1:

**Pull request closed without merging**